conduit/transforms/audio.py

```diff
@@ -0,0 +1,352 @@
+            onesided=onesided,
+            norm=norm,
+            mel_scale=mel_scale,
+        )
+        self.log_offset = log_offset
+
+    def forward(self, waveform: torch.Tensor) -> torch.Tensor:
+        x = super().forward(waveform)
+        return torch.log(x + self.log_offset)
+
+
+class LogMelSpectrogramNp(nn.Module):
+    # Mel spectrum constants and functions.
+
+    def __init__(
+        self,
+        sample_rate: int = 16000,
+        stft_window_length_seconds: float = 0.025,
+        stft_hop_length_seconds: float = 0.010,
+        log_offset: float = 0.01,  # Offset used for stabilized log of input mel-spectrogram.
+        num_mel_bins: int = 64,  # Frequency bands in input mel-spectrogram patch.
+        mel_min_hz: float = 125,
+        mel_max_hz: float = 7500,
+        mel_break_freq_hertz: float = 700.0,
+        mel_high_freq_q: float = 1127.0,
+    ):
+        super().__init__()
+        self.window_length_samples = int(round(sample_rate * stft_window_length_seconds))
+        self.hop_length_samples = int(round(sample_rate * stft_hop_length_seconds))
+        self.fft_length = 2 ** int(np.ceil(np.log(self.window_length_samples) / np.log(2.0)))
+        self.log_offset = log_offset
+        self.audio_sample_rate = sample_rate
+        self.num_mel_bins = num_mel_bins
+        self.lower_edge_hertz = mel_min_hz
+        self.upper_edge_hertz = mel_max_hz
+        self.mel_break_freq_hertz = mel_break_freq_hertz
+        self.mel_high_freq_q = mel_high_freq_q
+
+    def forward(self, data: torch.Tensor) -> torch.Tensor:
+        """Convert waveform to a log magnitude mel-frequency spectrogram.
+        Args:
+          data: 1D np.array of waveform data.
+          audio_sample_rate: The sampling rate of data.
+          log_offset: Add this to values when taking log to avoid -Infs.
+          window_length_secs: Duration of each window to analyze.
+          hop_length_secs: Advance between successive analysis windows.
+          **kwargs: Additional arguments to pass to spectrogram_to_mel_matrix.
+        Returns:
+          2D np.array of (num_frames, num_mel_bins) consisting of log mel filterbank
+          magnitudes for successive frames.
+        """
+        np_data = data.numpy()[0]
+        spectrogram = self.stft_magnitude(
+            np_data,
+            fft_length=self.fft_length,
+            hop_length=self.hop_length_samples,
+            window_length=self.window_length_samples,
+        )
+        mel_spectrogram = np.dot(
+            spectrogram,
+            self.spectrogram_to_mel_matrix(
+                num_spectrogram_bins=spectrogram.shape[1],
+                audio_sample_rate=self.audio_sample_rate,
+                num_mel_bins=self.num_mel_bins,
+                lower_edge_hertz=self.lower_edge_hertz,
+                upper_edge_hertz=self.upper_edge_hertz,
+            ),
+        )
+        return torch.Tensor(np.log(mel_spectrogram + self.log_offset), device=data.device)
```

Comment:
`torch.from_numpy`

Reply:
seems like the sort of thing that should be consistent

Reply:
`as_tensor` then